src/main/java/meli/freshfood/service/BatchServiceImpl.java

```diff
@@ -178,8 +213,28 @@ public void updateBatches(InboundOrderDTO inboundOrderDTO, Section section, Inbo
             return batch;
         }).collect(Collectors.toList());
 
-        batches.stream().forEach((batch)-> {
+        batches.stream().forEach((batch) -> {
             save(batch);
         });
     }
+
+    public List<Batch> sortByCategory(String storageType, List<Batch> batches) {
+        StorageType storageTypeObj = StorageType.parseToStorage(storageType);
+        return batches.stream().filter(b -> b.getProduct()
+                .getStorageType()
+                .equals(storageTypeObj))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<BatchStockDTO> findBatchesByCategoryAndDueDate(Integer intervalDate, String storageType, Boolean order) {
+        List<Batch> batches = batchRepository.findAll();
+        List<Batch> batchesSortedByCategory = this.sortByCategory(storageType, batches);
```

Comment:
Não é necessário o uso do "this", não é incorreto também, mas a gente poderia padronizar para deixar todos sem. Já que na maioria do código está usando sem. O que acha?